### PR TITLE
[Homework #18] - Hooks by Edelmiro Antón

### PR DESCRIPTION
**Pull request by EdelmiroAnton**
Hi @ilya-ischenko!
Here's my hw 18 about Hooks. 
Let me know if there's something to fix.

Thanks!

### CODE REVIEW

File: homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/Img.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import '../styles.css';
+import refresh from '../assets/refresh-button.svg';
+
+export const Img = ({ src, onClick }) => {
+  return (
+    <>
+      <img src={refresh} onClick={onClick} alt="refresh-button" className="refreshButton" />
+      <img src={src} alt="" className="avatar" />;
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/Img.js:9]: Remove ";" after img

File: homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/RefreshAllBtn.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import '../styles.css';
+
+export const RefreshAllBtn = ({ onClick }) => {
+  return (
+    <>
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/RefreshAllBtn.js:6]: <> is redundant

File: homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/App.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      setAvatar([...refresh]);
+    });
+    // console.log(...this.state.avatar);
+  };
+
+  const refreshAll = async () => {
+    const refreshAll = [...avatar];
+    const refreshAvatar = await Promise.all(refreshAll.map(() => fetchAvatar()));
+    setAvatar(refreshAvatar);
+  };
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/App.js:42]: Why do you need "refreshAll" const?

File: homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import React, { useState, useEffect } from 'react';
+import '../styles.css';
+import { AddButton } from './AddButton';
+import { Img } from './Img';
+import { RefreshAllBtn } from './RefreshAllBtn';
+
+export const App = () => {
+  const [avatar, setAvatar] = useState([]);
```
Comment (ilya-ischenko) [inline homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/App.js:8]: Avatar is not so good name for list of avatars

**Review by ilya-ischenko — COMMENTED** [review]
Remove console logs, please

File: homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/App.js (hunk 3 above)
Comment (EdelmiroAnton) [reply homeworks/edelmiro-anton/18-hooks/my-avatar/src/components/App.js:42]: I made it to make the code clearer but I realize that I can directly use the [. . .avatar] in the Promise.all: https://ibb.co/PM3ctKF

**Review by ilya-ischenko — COMMENTED** [review]
As far as I know App.js file should not be in the components folder. Move it to src, please, as well as it is one of the root components.

Add some functionality when data can not be fetched, for example we have bad url. I think alert with error message will be ok for now. 

You can try to add some loading states when we fetch image for example

**Comment by EdelmiroAnton** [comment]
Hey @ilya-ischenko! I've just added some loading functionality when the user clicks on the refresh all button. 
Let me know if there's something to improve.

Thanks in advance!

**Review by ilya-ischenko — APPROVED** [review]
Well done!
Approved.

**Pull request merged**